hydra-node/src/Hydra/API/Server.hs

```diff
@@ -27,7 +30,29 @@ data APIServerLog
   | APIResponseSent {sentResponse :: LByteString}
   | APIRequestReceived {receivedRequest :: LByteString}
   | APIInvalidRequest {receivedRequest :: LByteString}
-  deriving (Eq, Show)
+  deriving stock (Eq, Show, Generic)
+
+instance ToJSON APIServerLog where
```

Comment:
Do we really need the manual instance here given we have records for each constructor?

Reply:
The manual instance is there for 2 reasons:

1. There's no `ToJSON` instance for `PortNumber` (comes from `Network.Socket`, so we either need a newtype or an orphan)
2. There's no `ToJSON` instance for `ByteString`.

I am not _too happy_ with the `LByteString` in the logs though, currently rendering it as raw bytes even though, for valid inputs/outputs we **know** they are valid JSON. Right now, the logs look like:

```json
{"contents":{"sentResponse":"{\"parties\":[42,51],\"output\":\"readyToCommit\"}","tag":"APIResponseSent"}
```

(notice the escaped quotes on the 'sentResponse', this is actually the `LByteString` being rendered). Perhaps we could change the data-type from `LByteString` to `Either Text Aeson.Value`, and reserve `Left` for logging invalid inputs/outputs and `Right` for the valid ones. Then we can log valid values as plain JSON and, generically derive the JSON instance for APIServerLog (provided we also address the first point).

Reply:
Yes, having nested escaped JSON is a bit ugly. Something to think about later, I would say.
As for the `PortNumber`: We are already providing some beefed-up version in `Hydra.Network` so maybe it won't be such a stretch to add To/FromJSON instances

Reply:
If we have `Either Text Value` then we might as well have a only `Value`.

Reply:
Fixed in: https://github.com/input-output-hk/hydra-poc/pull/39/commits/f3bdbb44d9cd4873818aeaa76afcf7bba0c3996a